core/src/main/java/com/microsoft/applicationinsights/internal/util/MapUtil.java

```diff
@@ -35,14 +35,20 @@
  */
 public class MapUtil
 {
+    /**
+     * Copies entries from the source map to the target map, overwrites any values in target.
+     * Filters out null values if target is a {@link ConcurrentHashMap}.
+     * @param source the source map. Cannot be null.
+     * @param target the target map. Cannot be null.
+     * @param <Value> The type of the values in both maps
+     * @throws IllegalArgumentException if either {@code source} or {@code target} are null.
+     */
     public static <Value> void copy(Map<String, Value> source, Map<String, Value> target) {
         if (source == null) {
-            Preconditions.checkArgument(source != null, "source must not be null");
-            return;
+            throw new IllegalArgumentException("source must not be null");
```

Comment:
Generally, we don't want the customer's app to crash because of exception in tracing library. Now I do understand that this would happen most likely because of users fault and in that case they would like to know immediately. But I would like to discuss this a bit more. May be we can add try catch and call trackException() to report to AI?

Reply:
Look into `Preconditions.checkArgument`. This doesn't change how the method behaves.

Reply:
Alright makes sense. Thanks for explanation.